src/interpreter/executors/instruction.rs

```diff
@@ -496,5 +490,39 @@ where
     }
 }
 
+/// Computes logarithm for given exponent and base.
+/// Diverges when exp == 0 or base <= 1.
+#[inline(always)] // Force copy of each invocation for optimization, see comments below
+const fn _unchecked_ilog_inner(exp: Word, base: Word) -> u32 {
+    let mut n = 0;
+    let mut r = exp;
+    while r >= base {
+        r /= base;
+        n += 1;
+    }
+    return n;
+}
+
+/// Logarithm for given exponent and an arbitrary base, rounded
+/// rounded down to nearest integer value.
+///
+/// Must not be called if the exponent == 0, or if the base <= 1,
+/// as that causes either division by zero, or an infinite loop.
+///
+/// TODO: when https://github.com/rust-lang/rust/issues/70887 is stabilized,
+/// consider using that instead.
+const fn unchecked_ilog(exp: Word, base: Word) -> u32 {
```

Comment:
Why do we want this to be unchecked? Is there an efficient way to make this checked?

Reply:
see: https://github.com/FuelLabs/fuel-vm/issues/170

Reply:
This is unchecked due to the check already being done on callsite, where we use `alu_error` to check for the invariants. However, the compiler should be able to eliminate the `.expect(...)`, so I've changed this to `checked_` instead.